packages/modules/vehicles/vwid/api.py

```diff
@@ -1,93 +1,159 @@
-
-        data = await w.get_status()
-        if (data):
-            soc = (data['data']['batteryStatus']['currentSOC_pct'])
-            range = float(data['data']['batteryStatus']['cruisingRangeElectric_km'])
-            try:
-                f = open(replyFile, 'w', encoding='utf-8')
-            except Exception as e:
-                log.debug("vehicle " + str(vehicle) +
-                          ", replyFile open exception: e=" + str(e) + "user: " + getpass.getuser())
-                log.debug("vehicle " + str(vehicle) + ", replyFile open Exception, remove existing file")
-                os.system("sudo rm " + replyFile)
-                f = open(replyFile, 'w', encoding='utf-8')
-            json.dump(data, f, ensure_ascii=False, indent=4)
-            f.close()
+
+class api:
+
+    def __init__(self):
+        self.su = socUtils()
+        pass
+
+    # async method, called from sync fetch_soc, required because libvwid expects  async enviroment
+    async def _fetch_soc(self,
+                         conf: VWId,
+                         vehicle: int) -> Union[int, float]:
+        self.user_id = conf.configuration.user_id
+        self.password = conf.configuration.password
+        self.vin = conf.configuration.vin
+        self.refreshToken = conf.configuration.refreshToken
+        self.replyFile = 'soc_vwid_reply_vh_' + str(vehicle)
+        self.accessTokenFile = str(RAMDISK_PATH) + '/soc_vwid_accessToken_vh_' + str(vehicle)
+        self.accessToken_old = {}
+
+        async with aiohttp.ClientSession() as self.session:
+            self.w = libvwid.vwid(self.session)
+            self.w.set_vin(self.vin)
+            self.w.set_credentials(self.user_id, self.password)
+            self.w.tokens = {}
+            self.w.headers = {}
+
+            # initialize refreshToken
             try:
-                os.chmod(replyFile, 0o777)
-            except Exception as e:
-                log.debug("vehicle " + str(vehicle) + ", chmod replyFile exception, e=" + str(e))
-                log.debug("vehicle " + str(vehicle) + ", use sudo, user: " + getpass.getuser())
-                os.system("sudo chmod 0777 " + replyFile)
-
-            tokens_new = pickle.dumps(w.tokens)
-            if (tokens_new != tokens_old):    # check for modified tokens
-                log.debug("vehicle " + str(vehicle) + ", tokens_new != tokens_old, rewrite tokens file")
-                tf = open(tokensFile, "wb")
-                pickle.dump(w.tokens, tf)     # write tokens file
-                tf.close()
+                if self.refreshToken is None:
+                    log.debug("set refreshToken to initial value")
+                    self.w.tokens['refreshToken'] = initialToken
+                else:
+                    self.w.tokens['refreshToken'] = self.refreshToken
+
+            except Exception:
+                log.debug("refreshToken initialization exception: set refreshToken_old to initial value")
```

Comment:
```suggestion
                log.exception("refreshToken initialization exception: set refreshToken_old to initial value")
```
Bei Level Error/Exception werden die Fehler in jedem Fall geloggt. Wird der Fehler als Level Exception geloggt, ergänzt das Python-Logging automatisch Exception-Infos. (Level Exception darf nur im except-Zweig verwendet werden)
(betrifft auch weitere Exceptions in dieser Datei)